### ISSUE TITLE
What is the difference between Java build pack versions 3.X and 4.X?

### ISSUE THREAD

**Issue by dong-0kim** [cloudfoundry/java-buildpack#488]
Hi guys~
I am curious about the differences between the Java pack versions.
Version 4.5 adds CA Introscope APM Frameworks, a jvmkill Agent and a Monitoring Agent.
Are there any other differences?

**Comment by youngm** [comment]
You can see all the details here: https://github.com/cloudfoundry/java-buildpack/releases

Biggest change was to memory management detailed in this blog: https://www.cloudfoundry.org/just-released-java-buildpack-4-0/

**Comment by nebhale** [comment]
As @youngm points out, the biggest differences between 3.x and 4.x are documented in the [release announcement](https://www.cloudfoundry.org/just-released-java-buildpack-4-0/).  For a period of time (3.16/4.0 to 3.19/4.3) there were parallel releases where each new bit of functionality was added to both the 3.x and 4.x branches.  After a multi-month migration period, we've stopped adding new features to the 3.x branch and will only be adding them to the 4.x branch going forward.
